pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -118,7 +126,35 @@ func MakeVirtualServices(ia v1alpha1.IngressAccessor, gateways map[v1alpha1.Ingr
 	if requiredGatewayCount > 0 {
 		vss = append(vss, MakeIngressVirtualService(ia, gateways))
 	}
-	return vss
+
+	// Add probe routes
+	for _, vs := range vss {
+		hash, err := computeVirtualServiceHash(vs)
```

Comment:
Are there options other than a hash?  Can we use parent object UUID, may be?  

I worry that we may be sensitive to ordering of items here.

Reply:
Hash is the simplest. Parent's UUID could be a problem when the output of Reconcile is not simply a deterministic function of the parent, for example when configmaps are in the picture.